### PR TITLE
Update reference generator for kubectl

### PR DESCRIPTION
**Pull request by tengqm**
This PR migrates the `kubectl` reference generator to use go modules. Since the migration of `kubectl` into staging is almost done, the generator doesn't need to import the `k8s.io/kubernetes` repo.

### CODE REVIEW

**Comment by kbhawkey** [comment]
Hold for review.
The changes seem OK.
With these changes, I am wondering if it is possible to use the `k8s/website` update-imported-docs script?
If not, what problems need fixing?
/hold

**Comment by tengqm** [comment]
@kbhawkey This PR only focus on simplifying the generators and adding more targets to CI jobs. I'll look at the website script and propose fixes if necessary.

**Comment by tengqm** [comment]
@kbhawkey Nothing needs to change from the website scripts for kubectl. This fix focuses on simplifying the generator.

File: gen-kubectldocs/go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module github.com/kubernetes-sigs/reference-docs/gen-kubectldocs
+
+go 1.15
```
Comment (kbhawkey) [inline gen-kubectldocs/go.mod:3]: nit: go 1.16?

Comment (tengqm) [reply gen-kubectldocs/go.mod]: Yup. 1.16 would be better.

Comment (tengqm) [reply gen-kubectldocs/go.mod]: Fixed.

**Comment by kbhawkey** [comment]
Sorry for the delay in reviewing.
I am assuming these changes are up-to-date.
/lgtm

**Pull request merged**